### PR TITLE
mgr/cephadm: still check agent deps if it is marked down

### PR DESCRIPTION
**Pull request by adk3798**
Fixes: https://tracker.ceph.com/issues/53723

Signed-off-by: Adam King <adking@redhat.com>

Additionally, it's important to make sure the agent actually successfully received the new
config before updating the deps so that has been moved to the thread that sends the config to the agent

Most of the line changes here are just un-indenting a bunch of code that was formerly part of an else block.

Thinking this will fix the teuthology issues in the linked tracker. Will have to make sure that `cephadm/workunits/{agent/on mon_election/connectivity task/test_nfs}` and `cephadm/workunits/{agent/on mon_election/connectivity task/test_orch_cli}` don't fail with wait until healthy timeouts on whatever qa runs this is put through.

## Checklist
- Tracker (select at least one)
  - [x] References tracker ticket
  - [ ] Very recent bug; references commit where it was introduced
  - [ ] New feature (ticket optional)
  - [ ] Doc update (no ticket needed)
  - [ ] Code cleanup (no ticket needed)
- Component impact
  - [ ] Affects [Dashboard](https://tracker.ceph.com/projects/dashboard/issues/new), opened tracker ticket
  - [x] Affects [Orchestrator](https://tracker.ceph.com/projects/orchestrator/issues/new), opened tracker ticket
  - [ ] No impact that needs to be tracked
- Documentation (select at least one)
  - [ ] Updates relevant documentation
  - [x] No doc update is appropriate
- Tests (select at least one)
  - [ ] Includes [unit test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/tests-unit-tests/)
  - [ ] Includes [integration test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/testing_integration_tests/)
  - [ ] Includes bug reproducer
  - [x] No tests

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test classic perf`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test dashboard cephadm`
- `jenkins test api`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`

</details>

### CODE REVIEW

**Comment by sebastian-philipp** [comment]
Could you add a more meaningful description to your commit message?

File: src/pybind/mgr/cephadm/agent.py [hunk 1]
```diff
@@ -329,6 +330,13 @@ def run(self) -> None:
                 agent_response = secure_agent_socket.recv(1024).decode()
                 self.mgr.log.debug(f'Received "{agent_response}" from agent on host {self.host}')
                 self.mgr.cache.sending_agent_message[self.host] = False
+                if self.daemon_spec:
+                    # agent successfully received new config. Update config/deps
+                    self.mgr.cache.update_daemon_config_deps(
+                        self.daemon_spec.host, self.daemon_spec.name(), self.daemon_spec.deps, datetime_now())
+                    self.mgr.cache.agent_timestamp[self.daemon_spec.host] = datetime_now()
+                    self.mgr.cache.agent_counter[self.daemon_spec.host] = 1
+                    self.mgr.cache.save_host(self.daemon_spec.host)
```
Comment (sebastian-philipp) [inline src/pybind/mgr/cephadm/agent.py:339]:
```suggestion
                self.mgr.cache.got_agent_response(host, daemon_spec)
```
Let's avoid 5 consecutive lines starting with `self.mgr.cache....` 
```

**Review by sebastian-philipp — CHANGES_REQUESTED** [review]
commit message is empty

**Comment by ljflores** [comment]
jenkins test api

**Comment by ljflores** [comment]
jenkins test cephadm

**Comment by ljflores** [comment]
jenkins test dashboard cephadm

**Review by sebastian-philipp — APPROVED** [review]

**Comment by sebastian-philipp** [comment]
https://pulpito.ceph.com/swagner-2022-01-17_12:42:04-orch:cephadm-wip-swagner-testing-2022-01-17-1014-distro-default-smithi/

* ingress is broken
* hardware issue on smithi187
* crun + cgroup issue

**Pull request merged**